### PR TITLE
Avoid negative epoch totals for unlimited training

### LINKED ISSUE #21925: `RichProgressBar` shows a negative epoch total (e.g. `Epoch 5/-2`) when `max_epochs=-1`
### Bug description

`RichProgressBar` shows an incorrect, negative epoch total when `Trainer(max_epochs=-1)` is used.

Normally, with a finite `max_epochs` (e.g. `max_epochs=10`), the bar shows something like `Epoch 5/10`. But `max_epochs=-1` is Lightning's convention for "no limit on epochs" — training is expected to run until some other stopping condition is met (e.g. `max_steps`). In that case, instead of just `Epoch 5`, the bar shows `Epoch 5/-2`.

Steps to reproduce:
1. Create a `Trainer` with `max_epochs=-1`, `max_steps=<N>`, and `callbacks=[RichProgressBar()]`.
2. Call `trainer.fit(...)` with a dataloader such that training runs for more than one epoch before `max_steps` is reached.
3. Observe the train progress bar description.

You can just fetch an example here: github.com/pupeno/pytorch-lightning-workspace/tree/main/scratch/21924_rich_progress_bar_infinite_epochs

Expected behavior: the bar shows just `Epoch 5` (no total), since the number of epochs is unbounded.

Actual behavior: the bar shows `Epoch 5/-2`.

Root cause: `RichProgressBar._get_train_description()` in `src/lightning/pytorch/callbacks/progress/rich_progress.py` only guards against `self.trainer.max_epochs is not None`, but `-1` is not `None`, so it falls through to `train_description += f"/{self.trainer.max_epochs - 1}"`, i.e. `-1 - 1 = -2`.

### What version are you seeing the problem on?

master

### Reproduced in studio

_No response_

### How to reproduce the bug

```python
import torch
from torch.utils.data import DataLoader, TensorDataset
from lightning.pytorch import LightningModule, Trainer
from lightning.pytorch.callbacks import RichProgressBar


class Model(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(1, 1)

    def training_step(self, batch, batch_idx):
        x, y = batch
        loss = torch.nn.functional.mse_loss(self.layer(x), y)
        self.log("loss", loss)
        return loss

    def configure_optimizers(self):
        return torch.optim.SGD(self.parameters(), lr=0.1)


bar = RichProgressBar()
trainer = Trainer(
    accelerator="cpu",
    devices=1,
    max_epochs=-1,  # "no limit on epochs"
    max_steps=6,
    callbacks=[bar],
    num_sanity_val_steps=0,
    enable_checkpointing=False,
    enable_model_summary=False,
    logger=False,
)
x = torch.tensor([[1.0], [2.0], [3.0]])
y = torch.tensor([[2.0], [4.0], [6.0]])
train_dataloader = DataLoader(TensorDataset(x, y), batch_size=1)
trainer.fit(Model(), train_dataloaders=train_dataloader)

# Prints the exact description string used for each epoch's progress bar.
for epoch in range(trainer.current_epoch + 1):
    print(bar._get_train_description(epoch))
```

### Error messages and logs

```
Epoch 0/-2
Epoch 1/-2
Epoch 2/-2
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- PyTorch Lightning Version: master (commit 53cb84c49b78e81f286d761150839f5ecbe2f5b7)
#- PyTorch Version: 2.9.1+cu128
#- Python version: 3.12.14
#- OS: Ubuntu 26.04.1 LTS
#- CUDA/cuDNN version: N/A (reproduced on CPU)
#- GPU models and configuration: N/A (reproduced on CPU)
#- How you installed Lightning: source (editable install)
```

</details>

### More info

_No response_

cc @ethanwharris

### PR DESCRIPTION
**Pull request by pupeno**
## What does this PR do?

Fixes: #21925.

`RichProgressBar` shows an epoch total next to the current epoch, e.g. `Epoch 5/10`. When `Trainer(max_epochs=-1)` is used (Lightning's convention for "no limit on epochs") and training instead stops due to another condition (e.g. `max_steps`), the bar incorrectly displayed a nonsensical negative total, e.g. `Epoch 5/-2`.

The root cause is in `RichProgressBar._get_train_description()`: it only checked `self.trainer.max_epochs is not None` before appending `f"/{self.trainer.max_epochs - 1}"`. Since `-1` is not `None`, it was treated as a real epoch count and `-1 - 1` produced `-2`. The fix adds a check for `self.trainer.max_epochs >= 0`, so the total is omitted entirely when epochs are unlimited, and the bar now shows a plain `Epoch 2`.

No breaking changes.

<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary) — not needed, no documented/public behavior changed beyond the display fix.
- [x] Did you write any **new necessary tests**? (not for typos and docs) — yes, added `test_rich_progress_bar_infinite_epochs`.
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request? — none.
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors) — yes.

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

**Review by ethanwharris — APPROVED** [review]

**Comment by pupeno** [comment]
Thank you for the review @ethanwharris.

**Review by bhimrazy — APPROVED** [review]
